lib/diplomat/rest_client.rb

```diff
@@ -173,6 +170,88 @@ def use_consistency(options)
       options && options[:consistency] ? [options[:consistency].to_s] : []
     end
 
+    # rubocop:disable MethodLength, AbcSize, CyclomaticComplexity, PerceivedComplexity
+    # TODO: Migrate all custom params in options
+    def parse_options(options)
+      headers = nil
+      query_params = []
+      url_prefix = nil
+      consistency = []
+
+      headers = { 'X-Consul-Token': configuration.acl_token } if configuration.acl_token
+
+      unless options.nil?
+        # Parse options used as header
+        headers = { 'X-Consul-Token': options[:token] } if options[:token]
+
+        # Parse options used as query params
+        consistency = 'stale' if options[:stale]
+        consistency = 'leader' if options[:leader]
+        consistency = 'consistent' if options[:consistent]
+        query_params << consistency
+
+        # Parse url host
+        url_prefix = options[:http_addr] if options[:http_addr]
+      end
+      { query_params: query_params, headers: headers, url_prefix: url_prefix }
+    end
+    # rubocop:enable MethodLength, AbcSize, CyclomaticComplexity, PerceivedComplexity
+
+    # rubocop:disable MethodLength, AbcSize, CyclomaticComplexity
+    def send_get_request(connection, url, options, custom_params = nil)
+      rest_options = parse_options(options)
+      url += rest_options[:query_params]
+      url += custom_params unless custom_params.nil?
+      begin
+        connection.get do |req|
+          req.url rest_options[:url_prefix] ? rest_options[:url_prefix] + concat_url(url) : concat_url(url)
+          rest_options[:headers].map { |k, v| req.headers[k.to_sym] = v } unless rest_options[:headers].nil?
+          req.options.timeout = options[:timeout] if options && options[:timeout]
+        end
+      rescue Faraday::ClientError => e
+        raise Diplomat::PathNotFound, e
+      end
+    end
+    # rubocop:enable MethodLength, AbcSize, CyclomaticComplexity
+
+    # rubocop:disable MethodLength, AbcSize, CyclomaticComplexity
+    def send_put_request(connection, url, options, data, custom_params = nil)
```

Comment:
OPT: using named arguments would make calls to this method easier to read

Reply:
I'll plan this for another review